lib/api/core/plugins/pluginsManager.js

```diff
@@ -243,6 +243,30 @@ class PluginsManager {
   listStrategies () {
     return Bluebird.resolve(this.registeredStrategies);
   }
+
+  /**
+   * Stops and unregisters plugins workers
+   * @returns {Promise}
+   */
+  shutdownWorkers() {
+    const names = Object.keys(this.workers);
+
+    if (names.length === 0) {
+      return Bluebird.resolve();
+    }
+
+    return new Bluebird(resolve => {
+      names.forEach(name => {
+        this.workers[name].pmIds.getArray().forEach(id => pm2.delete(id));
+      });
+
+      this.workers = {};
+
+      // gives time for pm2 to delete plugins before resolving
+      // the promise
+      setTimeout(() => resolve(), 100);
```

Comment:
nitpicking: 100ms sounds very short to me depending on the plugins complexity. I would rather make it configurable, or __better__ monitor the pids and retry till they are up.

Further, using `Bluebird.delay` instead of a Promise constructor + setTimeout here would be a little clearer imo.